### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by cyb17**
combinaison auth

### CODE REVIEW

**Review by elrichardeau — APPROVED** [review]

**Review by juandrie — APPROVED** [review]

File: backend/auth/authentication/serializers.py [hunk 1]
```diff
@@ -6,7 +6,21 @@ class Meta:
         model = User
         fields = ('id', 'username', 'nickname', 'email', 'avatar', 'is_active', 'is_superuser', 'is_staff', 'friends', 'password')
         read_only_fields = ['is_superuser']
-        extra_kwargs = {'password': {'write_only': True}}
+        extra_kwargs = {
+            'id': {'read_only': True},
```
Comment (mortebrume) [inline backend/auth/authentication/serializers.py:10]: unecessary kwargs here, read only can be put on read_only_fields

File: backend/auth/authentication/models.py [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 from .validators import validate_alnum
 from .manager import UserManager
 
 
 class User(AbstractUser, PermissionsMixin):
 	username = models.CharField(max_length=15, unique=True, validators=[MinLengthValidator(3), validate_alnum])
-	nickname = models.CharField(max_length=30)
-	email = models.EmailField(unique=True)
-	avatar = models.URLField(blank=True, null=True)
+	nickname = models.CharField(max_length=30, blank=True, null=True)
```
Comment (mortebrume) [inline backend/auth/authentication/models.py:10]: blank false, nickname is not mandatory

File: backend/auth/authentication/serializers.py [hunk 3]
```diff
@@ -6,7 +6,21 @@ class Meta:
...
+			'friends': {'read_only': True},
+			'is_staff': {'read_only': True},
+			'is_active': {'read_only': True},
+			'date_joined': {'read_only': True},
+            'password': {'write_only': True}
+            }
+
+    # save() par defaut ne hash pas le MDP user, donc redefinition ici.
+    def save(self, **kwargs):
```
Comment (mortebrume) [inline backend/auth/authentication/serializers.py:19]: use manager here, the password is hashed twice, the version before was adequate

File: backend/auth/authentication/views.py [hunk 4]
```diff
@@ -12,9 +13,57 @@
...
+            return True
+        # Si ce n'est pas un admin/staff, il n'a que des droits de lecture (GET, HEAD, OPTIONS)
+        return request.method in permissions.SAFE_METHODS
+
 class UserViewSet(viewsets.ModelViewSet):
     queryset = User.objects.all()  # Récupère tous les utilisateurs
     serializer_class = UserSerializer  # Utilise le serializer pour convertir les données
+    permission_classes = [IsAccountAdminOrReadOnly]
+
+    @action(detail=False, methods=['post'], permission_classes=[AllowAny], url_path='register')
```
Comment (mortebrume) [inline backend/auth/authentication/views.py:29]: i think django rest provides a simpler option, you'll need to check

**Review by mortebrume — APPROVED** [review]
LGTM, assigning @elrichardeau for the followups comments to be addressed on a later PR

**Pull request merged**